sanic/testing.py

```diff
@@ -165,3 +173,197 @@ def head(self, *args, **kwargs):
 
     def websocket(self, *args, **kwargs):
         return self._sanic_endpoint_test("websocket", *args, **kwargs)
+
+
+class SanicASGIAdapter(requests.asgi.ASGIAdapter):
+    async def send(  # type: ignore
+        self,
+        request: requests.PreparedRequest,
+        gather_return: bool = False,
+        *args: typing.Any,
+        **kwargs: typing.Any,
+    ) -> requests.Response:
+        scheme, netloc, path, query, fragment = urlsplit(
+            request.url
+        )  # type: ignore
+
+        default_port = {"http": 80, "ws": 80, "https": 443, "wss": 443}[scheme]
```

Comment:
👍

Reply:
On closer inspection, I think I am going to leave this. I am adding a note about this.

This code is *almost* entirely from the inherited class: https://github.com/encode/requests-async/blob/614f40f77f19e6c6da8a212ae799107b0384dbf9/requests_async/asgi.py#L51

The difference is that we need to be able to have a `return_value` on the `response` that is the return of the ASGI method call. Why? Because it was the best way that I could come up with to maintain the `(request, response)` tuple from the test client call. I am playing around with some different options for a PR to the underlying repo so our override is much smaller. But, there is a deep set of variable scoping that needs to be dealt with, so I have not done it yet.